src/main/java/com/tikal/jenkins/plugins/multijob/views/MultiJobView.java

```diff
@@ -145,63 +193,163 @@ private void addProjectFromBuilder(MultiJobProject project,
-							jobBuildState.lastBuildNumber, lastSuccess, lastFailure);
+                BuildState jobBuildState = createBuildState(
+                        buildState,
+                        project,
+                        null,
+                        projectConfig
+                );
+                // phaseWrapper.addChildBuildState(jobBuildState);
+                addSimpleProject(
+                        project,
+                        subProject,
+                        jobBuildState,
+                        phaseNestLevel + 1,
+                        out,
+                        null
+                );
+            }
+        }
+    }
+
+    @SuppressWarnings("rawtypes")
+    private void addProjectFromBuilder(MultiJobBuild build,
+            BuildState buildState, List<TopLevelItem> out, BuildStep builder,
+            int phaseNestLevel, boolean isConditional) {
+        MultiJobBuilder reactorBuilder = (MultiJobBuilder) builder;
+        List<PhaseJobsConfig> subProjects = reactorBuilder.getPhaseJobs();
+        String currentPhaseName = reactorBuilder.getPhaseName();
+        PhaseWrapper phaseWrapper = new PhaseWrapper(
+                build.getProject(),
+                phaseNestLevel,
+                currentPhaseName,
+                isConditional
+        );
+        out.add(phaseWrapper);
+
+        for (PhaseJobsConfig projectConfig : subProjects) {
+            Item abstractProject = Jenkins.getInstance().getItem(
+                    projectConfig.getJobName(),
+                    build.getParent(),
+                    AbstractProject.class
+            );
+            if (abstractProject instanceof MultiJobProject) {
+                MultiJobBuild subBuild = null;
+                for (SubBuild sb : build.getSubBuilds()) {
+                    if (!(projectConfig.getJobName().equals(sb.getJobName()) &&
+                            projectConfig.getJobAlias().equals(sb.getJobAlias()))) {
+                        continue;
+                    }
+                    subBuild = (MultiJobBuild)sb.getBuild();
+                    break;
+                }
+                if (subBuild == null) {
+                    BuildState jobBuildState = createBuildState(
+                            buildState,
+                            build.getProject(),
+                            null,
+                            projectConfig
+                    );
+                    // phaseWrapper.addChildBuildState(jobBuildState);
+                    addMultiProject(
+                            build.getProject(),
+                            (MultiJobProject)abstractProject,
+                            jobBuildState,
+                            phaseNestLevel + 1,
+                            currentPhaseName,
+                            out
+                    );
+                } else {
+                    BuildState jobBuildState = createBuildState(
```

Comment:
This duplicates `if (subBuild == null)` branch.
That could you move `BuildState jobBuildState` above?